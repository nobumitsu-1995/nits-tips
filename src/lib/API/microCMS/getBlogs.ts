import type { MicroCMSQueries } from 'microcms-js-sdk'
import type { MicroCMS } from '@/types/microCMS'
import { getMicroCMSContents } from './client'

type Response = {
  contents: MicroCMS['blog'][]
}

const ENDPOINT = 'blogs'

export const getBlogsData = async (query?: MicroCMSQueries) => {
  const blogs = await getMicroCMSContents<Response | undefined>({
    endpoint: ENDPOINT,
    queries: {
      orders: '-publishedAt',
      limit: 15,
      ...query,
    },
  })

  if (!blogs) return undefined

  return blogs.contents
}
