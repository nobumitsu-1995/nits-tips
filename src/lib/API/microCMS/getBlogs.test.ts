import { getBlogsData } from './getBlogs'

const mockedJson = vitest.fn().mockResolvedValue({})
const mockedFetch = vitest.fn()

beforeEach(() => {
  const mockResponse = {
    ok: true,
    status: 200,
    json: mockedJson,
  }
  global.fetch = mockedFetch.mockResolvedValue(mockResponse)
})

afterEach(() => {
  vitest.resetAllMocks()
})

test('getBlogsで正常に値が取得できる', async () => {
  await getBlogsData({
    filters: 'isHidden[not_equals]true',
    orders: 'updatedAt',
  })
  expect(mockedFetch).toMatchSnapshot()
  expect(mockedJson).toHaveBeenCalledWith()
})
