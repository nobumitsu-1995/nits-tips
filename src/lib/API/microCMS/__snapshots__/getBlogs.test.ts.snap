// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getBlogsで正常に値が取得できる 1`] = `
[MockFunction spy] {
  "calls": [
    [
      "https://nits-tips.microcms.io/api/v1/blogs?orders=updatedAt&limit=15&filters=isHidden%5Bnot_equals%5Dtrue",
      {
        "headers": Headers {
          Symbol(entries): {
            "x-microcms-api-key": {
              "name": "X-MICROCMS-API-KEY",
              "value": [
                "sample",
              ],
            },
          },
        },
        "method": "GET",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": {
        "json": [MockFunction spy] {
          "calls": [
            [],
          ],
          "results": [
            {
              "type": "return",
              "value": {},
            },
          ],
        },
        "ok": true,
        "status": 200,
      },
    },
  ],
}
`;
